Makefile

```diff
@@ -1,32 +1,38 @@
-#http://www.gnu.org/prep/standards/html_node/Standard-Targets.html#Standard-Targets
-TOOL_ROOT?=$(shell pwd)/lib/binding
-OSRM_DATASTORE:=$(TOOL_ROOT)/osrm-datastore
-export TOOL_ROOT
-export OSRM_DATASTORE
+# http://www.gnu.org/prep/standards/html_node/Standard-Targets.html#Standard-Targets
 
-all: build
+all: build/Release/osrm.node
 
-pkgconfig:
-	@if [[ `which pkg-config` ]]; then echo "Success: Found pkg-config"; else "echo you need pkg-config installed" && exit 1; fi;
+ifeq (, $(shell which pkg-config))
+ $(error "No pkg-config found: please ensure you have pkg-config installed")
+endif
 
-./node_modules/node-pre-gyp:
+./node_modules/nan:
```

Comment:
@TheMarex This is not necessary, but an optimization. The goal of the `Makefile` wrapper is to speed up development when only one thing is expected to change: `src/node-osrm.cpp`. Running `npm install` and `make` should have the same result but the former incurs several seconds of wasted time while npm checks the dependency tree and does other fancy stuff and prints thingsto the terminal that are unrelated to C++ development on `src/node-osrm.cpp`. Therefore the Makefile has these shortcuts to quickly avoid re-running `npm install` when it looks like it is unneeded. This is not failsafe: if the first npm install works to install nan but fails to grab another dependency due to some network blip then you might need to do `make clean && make` to get the build working.